src/dmd/expressionsem.d

```diff
@@ -11746,3 +11772,25 @@ VarDeclaration makeThis2Argument(const ref Loc loc, Scope* sc, FuncDeclaration f
     vthis2.nestedrefs.push(fd);
     return vthis2;
 }
+
+/*******************************
+ * Make sure that the runtime hook exists.
+ * Params:
+ *      loc = location to use for error messages
+ *      sc = current scope
+ *      id = the hook identifier
+ *      description = what the hook does
+ *      module_ = what module the hook is located in
+ * Returns:
+ *      If the hook was found.
+ */
+extern (C++) bool verifyHookExist(const ref Loc loc, Scope* sc, Identifier id, const(char)* description, Identifier module_ = Id.object)
```

Comment:
Favor `ref Scope sc` if possible.

Reply:
Also, favor `string` instead of `const(char)*`.  We're trying to get rid of `char*` in DMD.

Reply:
> Favor `ref Scope sc` if possible.

Scope is stored as a pointer inside ExpressionSemanticVisitor.
I could change it to a ref, but then I need to dereference it each time `verifyHookExist` is called
Should I still do it?

> Also, favor `string` instead of `const(char)*`. We're trying to get rid of `char*` in DMD.

The `const(char)*` is sent all the way down to the C function `vsnprintf` via `OutBuffer`, so it need to be a `const(char)*`. I can't find a `tostringz` function so I could not got that way either.

Reply:
> Should I still do it?

Reply:
> I can't find a tostringz function so I could not got that way either.

Is it possible for you to use `f("%.*s", cast(int)s.length, s.ptr)` to workaround the current limitations of using `string` in DMD?  That technique is used in several other places in DMD currently.

Reply:
Yes, `"%.*s"` will work.

Reply:
> I can't find a tostringz function so I could not got that way either.

There's one here:

https://github.com/dlang/dmd/blob/df6418f0dd4cfeb79129a717f1a1037ccf4c90ae/src/dmd/utils.d#L207